lib/hpke/src/certificate.cpp

```diff
@@ -54,6 +54,12 @@ struct DNSName
 
 using GeneralName = tls::var::variant<RFC822Name, DNSName>;
 
+struct ParsedX509Subject
```

Comment:
It's pretty gross to just handle some specific fields out of the `Name` struct.  How about something like the following:

```C++
enum struct NameType : int {
  organization = NID_organizationName,
  organizational_unit = NID_organizationalUnitName,
  country = NID_countryName,
  state_or_province_name = NID_stateOrProvinceName,
  common_name = NID_commonName,
  serial_number = NID_serialNumber,
};

using ParsedName = std::map<NameType, std::string>;
```

That way your parsing code can just set `name[nid] = parsed_value`.

Reply:
And for completeness, might as well do this for the issuer as well.